semgrep-core/src/core/ast/AST_generic_helpers.ml

```diff
@@ -65,6 +65,18 @@ let name_of_ids ?(name_typeargs = None) xs =
 
 let name_of_id id = Id (id, empty_id_info ())
 
+let name_of_dot_access e =
```

Comment:
Maybe update the comment in AST_generic.ml about DotAccess and reference name_of_dot_access as temporary hack to solve the issue.

Reply:
Done! Rewriting Naming_AST for this and other stuff could be a future tech-debt project. GHC e.g. also uses the "renaming" phase to "fix" things in the AST that cannot be decided during parsing (such as our `DotAccess` vs `IdQualified`).